dataactbroker/handlers/fileHandler.py

```diff
@@ -507,6 +552,9 @@ def checkGeneration(self, submission_id = None, file_type = None):
         """
         if submission_id is None or file_type is None:
             submission_id, file_type = self.getRequestParamsForGenerate()
+        # Check permission to submission
+        self.checkSubmissionById(submission_id, file_type)
```

Comment:
This is now in JIRA as https://federal-spending-transparency.atlassian.net/browse/DB-1037